base_decorator.rb

```diff
@@ -0,0 +1,11 @@
+class BaseDecorator
+  include Nameable
+
+  def initialize(nameable)
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
```

Comment:
- [ ] The `BaseDecorator` should also inherit from the `Nameable` class and please make sure to add an attribute accessor to the `nameable` instance variable

Reply:
Suggestion implemented